src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/PodTemplateStep.java

```diff
@@ -80,44 +106,52 @@ public String getLabel() {
     }
 
     @DataBoundSetter
-    public void setLabel(String label) {
+    public void setLabel(@CheckForNull String label) {
         this.label = Util.fixEmpty(label);
     }
 
-    public @CheckForNull String getName() {
+    @CheckForNull
+    public String getName() {
         return name;
     }
 
     @DataBoundSetter
-    public void setName(String name) {
+    public void setName(@CheckForNull String name) {
         this.name = Util.fixEmpty(name);
     }
 
+    @CheckForNull
     public String getNamespace() {
         return namespace;
     }
 
     @DataBoundSetter
-    public void setNamespace(String namespace) {
-        this.namespace = namespace;
+    public void setNamespace(@CheckForNull String namespace) {
+        this.namespace = Util.fixEmpty(namespace);
     }
 
+    @CheckForNull
     public String getCloud() {
         return cloud;
     }
 
     @DataBoundSetter
-    public void setCloud(String cloud) {
-        this.cloud = cloud;
+    public void setCloud(@CheckForNull String cloud) {
+        this.cloud = Util.fixEmpty(cloud);
     }
 
+    @CheckForNull
     public String getInheritFrom() {
         return inheritFrom;
     }
 
     @DataBoundSetter
-    public void setInheritFrom(String inheritFrom) {
-        this.inheritFrom = inheritFrom;
+    public void setInheritFrom(@CheckForNull String inheritFrom) {
+        if (inheritFrom != null && inheritFrom.equals(",default,")) { // Added to make KubernetesPipelineTest.runInPodNestedExplicitInherit() pass
+            this.inheritFrom = "";
```

Comment:
No, this looks all backwards. `""` means do not inherit. Null means inherit in the default way.

```suggestion
            this.inheritFrom = null;
```